src/ApparentHorizons/ComputeExcisionBoundaryVolumeQuantities.tpp

```diff
@@ -0,0 +1,173 @@
+  const auto& phi =
+      get<GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>>(src_vars);
+
+  if (target_vars->number_of_grid_points() !=
+      src_vars.number_of_grid_points()) {
+    target_vars->initialize(src_vars.number_of_grid_points());
+  }
+
+  using metric_tag = gr::Tags::SpatialMetric<3, TargetFrame>;
+  using inv_metric_tag = gr::Tags::InverseSpatialMetric<3, TargetFrame>;
+  using lapse_tag = ::Tags::TempScalar<0, DataVector>;
+  using shift_tag = ::Tags::TempI<1, 3, Frame::Inertial, DataVector>;
+  using spacetime_normal_vector_tag =
+      ::Tags::TempA<2, 3, Frame::Inertial, DataVector>;
+
+  // Additional temporary tags used for multiple frames
+  using inertial_metric_tag = gr::Tags::SpatialMetric<3, Frame::Inertial>;
+  using inertial_inv_metric_tag =
+      gr::Tags::InverseSpatialMetric<3, Frame::Inertial>;
+  using inertial_spatial_ricci_tag = gr::Tags::SpatialRicci<3, Frame::Inertial>;
+
+  // All of the temporary tags, including some that may be repeated
+  // in the target_variables (for now).
+  using temp_tags_list_no_ricci =
+      tmpl::list<metric_tag, inv_metric_tag, lapse_tag, shift_tag,
+                 spacetime_normal_vector_tag, inertial_metric_tag,
+                 inertial_inv_metric_tag>;
+  using full_temp_tags_list = std::conditional_t<
+      tmpl::list_contains_v<DestTagList,
+                            gr::Tags::SpatialRicci<3, TargetFrame>>,
+      tmpl::push_back<temp_tags_list_no_ricci, inertial_spatial_ricci_tag>,
+      temp_tags_list_no_ricci>;
+
+  // temp tags without variables that are already in DestTagList.
+  using temp_tags_list =
+      tmpl::list_difference<full_temp_tags_list, DestTagList>;
+  TempBuffer<temp_tags_list> buffer(get<0, 0>(psi).size());
+
+  // These are always temporaries.
+  auto& lapse = get<lapse_tag>(buffer);
+  auto& shift = get<shift_tag>(buffer);
+  auto& spacetime_normal_vector = get<spacetime_normal_vector_tag>(buffer);
+
+  // These may or may not be temporaries, depending on if they are asked for
+  // in target_vars.
+  auto& inertial_metric = *(get<inertial_metric_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inertial_inv_metric =
+      *(get<inertial_inv_metric_tag>(
+          target_vars, make_not_null(&buffer)));
+  auto& metric = *(get<metric_tag>(
+      target_vars, make_not_null(&buffer)));
+  auto& inv_metric = *(get<inv_metric_tag>(
+      target_vars, make_not_null(&buffer)));
+
+  // Actual computation starts here
+
+  gr::spatial_metric(make_not_null(&inertial_metric), psi);
+  // put determinant of 3-metric temporarily into lapse to save memory.
+  determinant_and_inverse(make_not_null(&lapse),
+                          make_not_null(&inertial_inv_metric), inertial_metric);
+
+  gr::shift(make_not_null(&shift), psi, inertial_inv_metric);
+  gr::lapse(make_not_null(&lapse), shift, psi);
+  gr::spacetime_normal_vector(make_not_null(&spacetime_normal_vector), lapse,
+                              shift);
```

Comment:
why does the moving mesh case compute so many more variables than the static mesh? I understand this is just copied from the AH, but maybe add some comments?

Reply:
I can remove the spacetime normal vector computation, this was needed for the computation of the extrinsic curvature but because it's not computed anymore I can remove it.